google/cloud/spanner_dbapi/connection.py

```diff
@@ -63,6 +63,7 @@ def __init__(self, instance, database):
 
         self.is_closed = False
         self._autocommit = False
+        self._use_mutations = False
```

Comment:
I think it would be better to allow three different values (so using something like an enum instead of boolean):
- Never
- OnlyForAutocommit (Default)
- Always

The user should have the option of turning the feature completely off, also when in autocommit, as we are trying to parse the SQL statements in `execute_many`. As that parsing is a best-effort feature, there will be cases when the parsing does not work as intended, and the user should have the option of turning the usage of mutations off.